src/webgpu/api/validation/encoding/beginComputePass.spec.ts

```diff
@@ -191,3 +191,35 @@ g.test('timestamp_query_set,device_mismatch')
 
     t.tryComputePass(!mismatched, descriptor);
   });
+
+g.test('begin_compute_pass_before_ending_previous_pass')
```

Comment:
I wrote this test referring to `CommandBufferValidationTest.BeginComputePassBeforeEndPreviousPass`[1] test. But, I'm not fully sure if this test was already implemented. wdyt?

[1] https://source.chromium.org/chromium/chromium/src/+/main:third_party/dawn/src/dawn/tests/unittests/validation/CommandBufferValidationTests.cpp;l=145?q=CommandBufferValidationTests.cpp&ss=chromium

Reply:
Took me a bit to find it but the file for this is `encoder_state.spec.ts`. It is currently empty. It has a very vague test plan though it's quite confusing. We had a draft issue open for it, so I just filed it and assigned you. #1914

Since we need many similar tests between render passes and compute passes, I do think it makes sense to put them all in one file, so I guess let's move this there.

Reply:
As mentioned in that issue there's possibility for parameterization by [compute, compute], [compute, render], [render, compute], etc. but we weren't sure if it would be best to just write separate tests for each one.

Reply:
yes, I also considered using parameters to test the cases of the combination of `compute` and `render` passes. But, it seemed like it would get complicated.